frontend/src/pages/AllRunsList.tsx

```diff
@@ -103,6 +116,40 @@ class AllRunsList extends Page<{}, AllRunsListState> {
       this.props.history.push(RoutePage.NEW_RUN + searchString);
     }
   }
+
+  private async _archiveDialogClosed(confirmed: boolean): Promise<void> {
+    if (confirmed) {
+      const unsuccessfulIds: string[] = [];
+      const errorMessages: string[] = [];
+      await Promise.all(this.state.selectedIds.map(async (id) => {
```

Comment:
should this be `forEach` since it's not actually returning anything?

Reply:
Then the API calls aren't made in parallel.

Reply:
Even though you're calling `await` within the map? Interesting

Reply:
Yes, AFAIK `Promise.all` is the only way to execute and wait on promises in parallel, otherwise the forEach -> async API call dispatches the next call after the first one is done.